### DISCUSSION TITLE
Missing business units in AR/AP search output

### DISCUSSION THREAD

**Discussion by ehuelsmann** [ledgersmb/LedgerSMB#5824] [Design]
On the AR/AP search page, many columns can be added to reports. However, quite a few of them aren't actually added to the output when the column is checked on the search screen.

**_Summarizing up to https://github.com/ledgersmb/LedgerSMB/issues/3997#issuecomment-699481029, the only thing missing now, are the business units._**

To resolve this item, we need to define the scope of the problem we're solving (do we need to be able to select multiple business units??) and a UI design to go with it.

**Comment by ehuelsmann** [comment]
Selectable columns, with those shown checked:

* [ ] No
* [x] Subtotal   `   <----- Not a column; it's a row instead`
* [ ] Business units `<- Not a regular column: an array, for which we don't have any support...`
* [x] Due date
* [x] Date paid
* [x] Order number
* [x] PO Number
* [x] Currency
* [x] Invoice date
* [x] Amount
* [x] Paid
* [x] Shipping point
* [x] ID
* [x] Tax
* [x] Ship Via
* [x] Invoice number
* [x] Customer
* [x] Total
* [x] Amount due
* [x] Sales person
* [x] Notes
* [x] Manager
* [x] Till

**Comment by ehuelsmann** [comment]
Columns of the search return type:

    id int,
    invoice bool,
    entity_id int,
    meta_number text,
    entity_name text,
    transdate date,
    invnumber text,
    ordnumber text,
    ponumber text,
    curr char(3),
    amount numeric,
    netamount numeric,
    tax numeric,
    paid numeric,
    due numeric,
    last_payment date,
    due_date date,
    notes text,
    till text,
    salesperson text,
    manager text,
    shipping_point text,
    ship_via text,
    business_units text[]

**Comment by ehuelsmann** [comment]
Columns configured in the report's/invoices/transactions: 

    return [
       { col_id => 'id',
           name => $self->Text('ID'),
           type => 'text'},
       { col_id => 'transdate',
           name => $self->Text('Date'),
           type => 'text'},
       { col_id => 'meta_number',
           name => $meta_number_label,
           type => 'text'},
       { col_id => 'entity_name',
           name => $entity_name_label,
       href_base =>'contact.pl?action=get&entity_class='.$self->entity_class,
           type => 'href', },
       { col_id => 'invnumber',
           name => $self->Text('Invoice'),
           type => 'href'},
       { col_id => 'netamount',
           name => $self->Text('Amount'),
           type => 'text'},
       { col_id => 'tax',
           name => $self->Text('Tax'),
           type => 'text'},
       { col_id => 'amount',
           name => $self->Text('Total'),
           type => 'text'},
       { col_id => 'paid',
           name => $self->Text('Paid'),
           type => 'text'},
       { col_id => 'due',
           name => $self->Text('Due'),
           type => 'text'},
       { col_id => 'last_payment',
           name => $self->Text('Date Paid'),
           type => 'text'},
       { col_id => 'due_date',
           name => $self->Text('Due Date'),
           type => 'text'},
       { col_id => 'notes',
           name => $self->Text('Notes'),
           type => 'text'},
       { col_id => 'till',
           name => $self->Text('Till'),
           type => 'text'},
       { col_id => 'salesperson',
           name => $self->Text('Salesperson'),
           type => 'text'},
       { col_id => 'manager',
           name => $self->Text('Manager'),
           type => 'text'},
       { col_id => 'shipping_point',
           name => $self->Text('Shipping Point'),
           type => 'text'},
       { col_id => 'ship_via',
           name => $self->Text('Ship Via'),
           type => 'text'},
    ];

**Comment by ehuelsmann** [comment]
Not mapped in the report column definition:

* [x] Currency (curr)
* [x] PO Number (ponumber)
* [x] Order number (ordnumber)

(With those checked now implemented in the report)

**Comment by ehuelsmann** [comment]
`col_duedate` instead of `col_due_date` in the report filter prevents selection of the column

**Comment by ehuelsmann** [comment]
Weirdly enough; there's a 'till' column. We've removed till support long ago. Maybe we should remove the 'till' column from this output?

**Comment by ehuelsmann** [comment]
Studying the AR/AP search code in 1.2, the report would be able to create "summary" output. In our case the summary output is only supported on true reports, not on the search functionality.

Conclusion: the "No" column selector should be removed (because it's supposed to hold the count of underlying summarized lines).

**Comment by ehuelsmann** [comment]
After the `No` and `Till` columns have been removed and the subtotals implemented, this isn't `bite-sized` anymore, because we don't have a pattern for the business units being an array which needs to be unpacked into columns.

**Comment by aungzwin10** [comment]
Column are missing in following reports also.
- AR/AP > Reports > Outstanding
- Order Entry > Reports > Sale Orders
- Order Entry > Reports > Purchase Orders
- Quotations > Reports > Quotations
- General Journal > Search (only Subtotal is not working)

**Comment by ehuelsmann** [comment]
@aungzwin10 lets create separate issues for the other reports with missing columns in line with the approach here: creating a list of missing columns in the output, matching that list with each layer below to check if they are simply missing from output or that they need lower-layer support.
